openapi-cli/src/main/java/io/ballerina/openapi/generators/service/RequestBodyGenerator.java

```diff
@@ -167,10 +188,18 @@ private TypeDescriptorNode getNodeForPayloadType(Components components, Map.Entr
         }
         literals.remove(literals.size() - 1);
         SeparatedNodeList<Node> expression = NodeFactory.createSeparatedNodeList(literals);
-        ListConstructorExpressionNode valueExpr = NodeFactory.createListConstructorExpressionNode(
-                createToken(SyntaxKind.OPEN_BRACKET_TOKEN), expression, createToken(SyntaxKind.CLOSE_BRACKET_TOKEN));
-        SpecificFieldNode specificFieldNode = NodeFactory.createSpecificFieldNode(null, mediaType,
-                createToken(SyntaxKind.COLON_TOKEN), valueExpr);
+        if (equalDataType.size() == 1) {
```

Comment:
Can this also be more than 1? if so, `equalDataTypes` would be a better choice for the name

Reply:
yeah, it can be, I will change the parameter name.